tests/test_onedrive.py

```diff
@@ -295,6 +296,53 @@ def test_root_event():
     assert odp._convert_to_event(non_root_event, "123") is not None
 
 
+def test_event_filter():
+    _, odp = fake_odp()
+
+    # root not set
+    assert not odp.root_path
+    event = Event(FILE, "", "", "", True)
+    assert odp._filter_event(event) == EventFilter.PROCESS
+    event = Event(DIRECTORY, "", "", "", False)
+    assert odp._filter_event(event) == EventFilter.PROCESS
+    assert odp._filter_event(None) == EventFilter.IGNORE
+
+    class MockSyncState:
+        def get_path(self, oid):
+            if oid == "in-root":
+                return "/root/path"
+            elif oid == "out-root":
+                return "/path"
+            else:
+                return None
+
+    # root set
+    odp._root_oid = "root_oid"
+    odp._root_path = "/root"
+    odp.sync_state = MockSyncState()
+
+    e = Event(FILE, "", "", "", False)
+    assert odp._filter_event(e) == EventFilter.IGNORE
+    e = Event(FILE, "in-root", "/root/path2", "hash", True)
+    assert odp._filter_event(e) == EventFilter.PROCESS
+    e = Event(FILE, "in-root", None, None, False)
+    assert odp._filter_event(e) == EventFilter.PROCESS
+    e = Event(FILE, "out-root", "/path2", "hash", True)
+    assert odp._filter_event(e) == EventFilter.IGNORE
+    e = Event(FILE, "out-root", None, None, False)
+    assert odp._filter_event(e) == EventFilter.IGNORE
+    e = Event(FILE, "in-root", "/path2", "hash", True)
+    assert odp._filter_event(e) == EventFilter.PROCESS
+    e = Event(FILE, "out-root", "/root/path2", "hash", True)
+    assert odp._filter_event(e) == EventFilter.PROCESS
+    e = Event(DIRECTORY, "out-root", "/root/path2", "hash", True)
+    assert odp._filter_event(e) == EventFilter.WALK
```

Comment:
Maybe I'm missing something but why isn't this a PROCESS? Isn't this a rename out of the root?

Reply:
Scratch. Reading again this is a rename INTO the root. LGTM